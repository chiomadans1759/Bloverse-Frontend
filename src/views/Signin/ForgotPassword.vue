<template>
  <div class="signin-container">
    <div class="top">
      <router-link to="/" class="router-link">
            <img class="logo" src="@/assets/Logo - White.svg" style="height: 40px" />
          </router-link>
    </div>
    <div class="content">
      <forgot-component></forgot-component>
    </div>
    <div class="bottom">
      <p><span class="bloverse">bloverse</span> 2018 All rights reserved</p>
    </div>
  </div>
</template>
<script>

import ForgotComponent from '@/components/SigninSection/forgot.vue';

export default {
  components: {
    ForgotComponent,
  }
}
</script>
<style lang="scss">
  @import './style.scss';
</style>
