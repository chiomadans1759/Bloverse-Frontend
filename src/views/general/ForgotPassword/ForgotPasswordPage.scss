@import 'sass-rem';

.forgot-password-container {
  padding-top: rem(15px);
  padding-bottom: rem(20px);
  min-height: 100vh;
  background-color: #333333;
  color: white;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  background-image:
        linear-gradient(rgba(37, 42, 48, 0.2), rgba(37, 42, 48, 0.2)),
        url('../../../assets/signin-image.jpg');
}

.forgot-password-content {
  min-height: calc(100vh - 170px);
  position: relative;
}

.forgot-password-intro-one {
  font-size: rem(24px);
  font-weight: 900;
  margin: 0 0 rem(8px) 0;
  width: rem(281px);
}

.forgot-password-intro-two {
  font-size: rem(16px);
  margin-bottom: rem(24px);
  width: rem(281px);
}

// These styles apply to other pages 
.password-grid-container {
  display: grid;
  grid-template-columns: 1fr 0.3fr 5fr;
}

.password-grid-item {
  opacity: .5;
}

.password-middle-grid-item {
  text-align: center;
  font-size: rem(16px);
  margin: -0.5rem 0 1.5rem 0;
}

.password-hr-text {
  line-height: 0em;
  position: relative;
  outline: 0;
  border: 0;
  text-align: left;
  height: rem(2em);
  opacity: .8;
  width: 100%;
  margin: -0.5rem 0 1.5rem 0;
}

.password-hr-text:before {
  content: '';
  background: #ad938a;
  position: absolute;
  top: 50%;
  width: 100%;
  height: rem(1px);
}

.password-or-text {
  padding: rem(10px);
}

.password-primary-button {
  background-image: linear-gradient( to left, #2F80ED, #003CE1);
  width: rem(358px);
  height: rem(38px);
  margin-bottom: rem(24px);
}

/* Extra small devices (phones, 600px and down) */
@media (min-width: 320px) and (max-width: 600px) { 
  .forgot-password-container {
    // background-color: #F5F5F5;
    // color: #252525;
    // background-image: none;
  }

  .password-primary-button {
    width: 19rem;
  } 
}
