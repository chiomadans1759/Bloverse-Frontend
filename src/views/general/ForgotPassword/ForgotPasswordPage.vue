<template>
  <div class="forgot-password-container">
    <navbar
      :showBackArrow="showBackArrow"
      :showNavigations="showNavigations"
      customClass="auth-nav"
      isTransparent
    />
    <div
      class="forgot-password-content d-flex flex-column justify-content-center align-items-center"
    >
      <div>
        <h4 class="forgot-password-intro-one">Forgot Password</h4>
        <h5 class="forgot-password-intro-two">Enter your email address and
          <br>we will send you a link to reset
          <br>your password
        </h5>
        <div class="forgot-password-form-section">
          <text-input
            id="email"
            name="email"
            type="email"
            placeholder="Email"
            :value="email"
            :onChange="onChange"
            custom-class="text-input"
          />
          <Button customClass="btn btn-primary password-primary-button">Send</Button>

          <div class="password-grid-container">
            <div class="password-grid-item">
              <hr class="password-hr-text">
            </div>
            <div class="password-middle-grid-item">
              <p>
                <strong class="password-or-text">Or</strong>
              </p>
            </div>
            <div class="password-grid-item">
              <hr class="password-hr-text">
            </div>
          </div>
        </div>
      </div>

      <Button customClass="btn btn-outline-secondary secondary-button">
        <h2 class="sign-up-new-bloverse-text text-center">Create
          <router-link to="/register" class="signup-login-text">new account</router-link>
        </h2>
      </Button>
    </div>
    <div class="text-center m-3">
      <p>
        <span class="bloverse">bloverse</span>
        {{year}}. All Right reserved
      </p>
    </div>
  </div>
</template>

<script>
import Button from "@/components/Button/Button.vue";
import TextInput from "@/components/TextInput/TextInput.vue";
import Navbar from "@/components/Navbar/Navbar.vue";

export default {
  data() {
    return {
      email: "",
      year: new Date().getFullYear(),
      showNavigations: false,
      showBackArrow: true
    };
  },
  components: {
    Button,
    TextInput,
    Navbar
  },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  }
};
</script>

<style lang="scss">
@import "./ForgotPasswordPage.scss";
</style>
