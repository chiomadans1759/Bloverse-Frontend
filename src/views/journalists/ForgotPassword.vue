<template>
  <main class="forgot-password">
    <div class="row justify-content-center pt-5">
      <div class="col-md-3 mt-5">
        <div :class="['alert', `alert-${alert.type}`]" v-show="alert.show">
          <b class="alert-heading">{{alert.msg}}</b>
        </div>

        <router-link to="/">
          <img src="@/assets/Logo.svg" style="width: 2.8rem;" alt class="img-logo mb-5">
        </router-link>

        <form>
          <h4 class="mb-4">Enter registered email address to recover password</h4>
          <div class="form-group">
            <input
              type="email"
              class="form-control"
              v-model="email"
              placeholder="e.g john.doe@mail.com"
            >
          </div>

          <div class="form-group">
            <button
              type="button"
              class="btn btn-primary btn-block"
              @click.prevent="resetPassword"
              :disabled="!email || processing"
            >Submit</button>
          </div>
        </form>
      </div>
    </div>
  </main>
</template>

<script>
import { mapActions } from "vuex";

export default {
  name: "forgot-password",
  data() {
    return {
      email: "",
      processing: false,
      alert: {
        show: false,
        type: null,
        code: null,
        msg: null
      }
    };
  },
  methods: {
    ...mapActions(["requestPasswordChange"]),

    async resetPassword() {
      let res = await this.requestPasswordChange(this.email);
      this.alert.show = true;
      if (res.statusCode == 404) {
        this.alert.type = "danger";
        this.alert.msg = "This is not a registered mail on the system.";
      } else if (res.statusCode == 200) {
        this.alert.type = "info";
        this.alert.msg =
          "Reset token has been sent successfully. Kindly check your mail.";
      }
      setTimeout(() => {
        this.alert.show = false;
      }, 10000);
    }
  }
};
</script>

<style scoped>
.forgot-password {
  background-color: #f5f5f5;
  border: 1px solid #eeeeee;
  height: 100vh;
  width: 100vw;
  overflow-x: hidden;
}
</style>

