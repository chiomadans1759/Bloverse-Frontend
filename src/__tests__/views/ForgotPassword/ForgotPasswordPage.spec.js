import { mount, RouterLinkStub } from '@vue/test-utils'
import ForgotPassword from '@/views/general/ForgotPassword/ForgotPasswordPage.vue';

describe('ForgotPassword Page', () => {
  it('should render correctly', () => {

    const wrapper = mount(ForgotPassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });


  it('should update the component state when input is changed', () => {
    const wrapper = mount(ForgotPassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const emailInput = wrapper.find('#email');
    emailInput.element.value = 'Janet@email.com';
    emailInput.trigger('change');
    expect(wrapper.vm.email).toBe('Janet@email.com');
  })
});
