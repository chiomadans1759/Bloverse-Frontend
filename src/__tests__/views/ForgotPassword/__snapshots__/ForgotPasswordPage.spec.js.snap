// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Page should render correctly 1`] = `
<div
  class="forgot-password-container"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5 transparent alt-button"
    >
      <div
        class="container-fluid auth-nav"
      >
        <a
          class="backlink mr-4"
        >
          <div
            class="btn rounded-btn-circle d-flex justify-content-center"
          >
            <i
              class="fal fa-arrow-left back-icon"
            />
          </div>
        </a>
         
        <span
          class="back-text"
        >
          Back
        </span>
         
        <a
          class="router-link"
        >
          <!---->
           
          <img
            class="nav-logo"
            src="@/assets/Bloverse-white.svg"
          />
        </a>
         
        <!---->
         
        <!---->
      </div>
    </nav>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Bug
      
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Comment
      
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Other
      
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                
              Close
            
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </header>
   
  <div
    class="forgot-password-content d-flex flex-column justify-content-center align-items-center"
  >
    <div>
      <h4
        class="forgot-password-intro-one"
      >
        Forgot Password
      </h4>
       
      <h5
        class="forgot-password-intro-two"
      >
        Enter your email address and
        
        <br />
        we will send you a link to reset
        
        <br />
        your password
      
      </h5>
       
      <div
        class="forgot-password-form-section"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control defaultClass text-input"
              id="email"
              name="email"
              placeholder="Email"
            />
          </div>
        </form>
         
        <div>
          <button
            class="btn btn-primary password-primary-button"
          >
            Send
          </button>
        </div>
         
        <div
          class="password-grid-container"
        >
          <div
            class="password-grid-item"
          >
            <hr
              class="password-hr-text"
            />
          </div>
           
          <div
            class="password-middle-grid-item"
          >
            <p>
              <strong
                class="password-or-text"
              >
                Or
              </strong>
            </p>
          </div>
           
          <div
            class="password-grid-item"
          >
            <hr
              class="password-hr-text"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <button
        class="btn btn-outline-secondary secondary-button"
      >
        <h2
          class="sign-up-new-bloverse-text text-center"
        >
          Create
        
          <a
            class="signup-login-text"
          >
            new account
          </a>
        </h2>
      </button>
    </div>
  </div>
   
  <div
    class="text-center m-3"
  >
    <p>
      <span
        class="bloverse"
      >
        bloverse
      </span>
      
      2019. All Right reserved
    
    </p>
  </div>
</div>
`;
